File: Stack,Queue/기능개발.py
import math

def solution(progresses, speeds):
    answer = [1]

    rest_progress = [math.ceil((100 - progresses[i])/speeds[i]) for i in range(len(progresses))]
    max_progress = rest_progress[0]
    for rp in rest_progress[1:]:
        if rp > max_progress:
            max_progress = max(rp, max_progress)
            answer.append(1)
        else:
            answer[-1] += 1

    return answer

print(solution([95, 90, 99, 99, 80, 99], [1, 1, 1, 1, 1, 1]))

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        Queue<Integer> stack = new LinkedList<>();
        
        for (int i = 0; i < progresses.length; i++) {
            int day = (100 - progresses[i]) / speeds[i];
            int rest = (100 - progresses[i]) % speeds[i];
            if (rest != 0) day++;
            stack.add(day);
        }
                
        List<Integer> dayList = new ArrayList<>();

        int work1 = stack.poll();
        int cnt = 1;
        while (!stack.isEmpty()) {
            int work2 = stack.peek();

            if (work1 >= work2) {
                cnt++;
                stack.poll();
            } else {
                dayList.add(cnt); // cnt 더한 것을 먼저 넣어줌 
                cnt = 1; // cnt 초기화
                work1 = stack.poll();
            }
        }
        dayList.add(cnt);
        int[] answer = new int[dayList.size()];
        for (int i = 0; i < answer.length; i++) {
            answer[i] = dayList.get(i);
        }
        
        return answer;
        
        // / 10 / 1 / 1 / 20 / 1
        
        // 1. stack 에 남은 기간들을 넣고
        // 뒤에것보다 앞의 것이 더 크면 pop 갯수
        
    }
}